### PR TITLE
Align and update tokens with Figma component library

### PR DESCRIPTION
**Pull request by steppy452**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features) 

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[x] Other... Please describe:  Shifting around some design tokens to align better with how our Figma component library works
```

## What is the current behavior?

Issue Number: N/A

```
{
  panel: {
    // Panel backgrounds, such as cards, tables, popovers, dialogs, dropdown menus, etc.
    DEFAULT: colorPalette['black-pearl'],
    content: colorPalette['athens-gray'],
    'secondary-content': colorPalette.gray[600],
    accent: colorPalette['charade']
  },
  surface: {
    // Form component backgrounds, such as text fields, checkboxes, select, etc.
    DEFAULT: colorPalette['charade'],
    content: colorPalette['athens-gray'],
    accent: colorPalette.blue[500],
    disabled: colors.gray[800]
  }
}
```

## What is the new behavior?

```

{
  panel: {
    // Panel backgrounds, such as cards, tables, popovers, dialogs, dropdown menus, etc.
    DEFAULT: colorPalette['black-pearl'],
    accent: colorPalette['charade']
  },
  surface: {
    // Form component backgrounds, such as text fields, checkboxes, select, etc.
    DEFAULT: colorPalette['charade'],
    accent: colorPalette.blue[500],
  },
  text: {
    'content-primary': colorPalette['athens-gray'],
    'content-secondary': colorPalette.gray[600]
  }
}
```

## Does this PR introduce a breaking change?
```
[x] Yes
[ ] No
```

Move `panel.content`, `panel.secondary-content` and `surface.content` tokens into a `text` block and renaming to `content-primary` and `content-secondary`

Theme files where `*-panel-content`, `*-panel-secondary-content` and `*-surface-content` are being used have been updated to `*-content-primary` and `*-content-secondary`. If existing applications `panel.content` and `surface.content` variables are different, updates to the following component theme files will need to be updated if the component is being used in the existing app:

- PagerTheme: `base`
- RedactTheme: `base`
- SortTheme: `base`
- ArrowTheme: `base`
- AvatarGroupTheme: `base`
- BadgeTheme:
  - `colors.primary`
  - `colors.secondary`
  - `colors.error`
- ButtonTheme:
  - `base`
  - `colors.default.text`
  - `colors.primary.filled`
  - `colors.secondary.filled`
  - `colors.success.filled`
  - `colors.warning.filled`
  - `colors.error.filled`
- DotsLoaderTheme: `dot`
- CheckboxTheme: `base`
- InputTheme:
  - `base`
  - `adornment.base`
- RangeTheme: `tooltip`
- SelectMenuTheme: `groupItem.title`
- TooltipTheme: `base`
- ListItemTheme:
  - `base`
  - `header`
- StackTheme: `base`
- TreeTheme:
  - `arrow`
  - `node.base`
  - `node.button.icon`

## Other information
Not affecting Reablocks theme, but `surface.disabled` was also removed and the story using the token was updated accordingly for both light and dark theme.

Figma plugin updates [here](https://github.com/reaviz/reablocks-figma-plugin/pull/5)

Migration notes will be updated as part of the [reablocks website PR](https://github.com/reaviz/reablocks-website/pull/31).

### CODE REVIEW

**Comment by steppy452** [comment]
@amcdnl after seeing some of the changes that Kyle is working on, I think it might be a good idea to try and take into account the changes that he's proposing before merging this.

File: docs/blocks/authentication/ForgotPassword.story.tsx [hunk 1]
```diff
@@ -355,7 +355,7 @@ export const ResetPassword = () => {
           </a>
         </Stack>
         <div className="w-full flex flex-col grow items-center justify-center text-center pt-2 mb-14">
-          <div className="p-3 border border-panel-secondary-content rounded mb-3">
+          <div className="p-3 border dark:border-waterloo light:border-gray-700 rounded mb-3">
```
Comment (amcdnl) [inline docs/blocks/authentication/ForgotPassword.story.tsx:358]: This couples the implementation to the specific color. Do we not have a abstraction for this?

File: docs/blocks/authentication/ForgotPassword.story.tsx [hunk 2]
```diff
@@ -61,7 +61,7 @@ export const ForgotPassword = () => {
           <h4 className="text-2xl font-sans font-bold mb-0">
             Reset your password
           </h4>
-          <span className="text-base text-panel-secondary-content font-sans">
+          <span className="text-base text-text-secondary font-sans">
```
Comment (amcdnl) [inline docs/blocks/authentication/ForgotPassword.story.tsx:64]: `text-text-secondary` - Not in love w/ that.

File: docs/blocks/authentication/ForgotPassword.story.tsx (hunk 1 above)
Comment (steppy452) [reply docs/blocks/authentication/ForgotPassword.story.tsx:358]: yea - i can probably use one of the accent colors. i'll update

**Pull request merged**